cg/services/orders/validation/order_types/tomte/models/sample.py

```
@@ -18,7 +18,7 @@ class TomteSample(Sample):
     phenotype_groups: list[str] | None = None
     phenotype_terms: list[str] | None = None
     post_formalin_fixation_time: int | None = None
-    reference_genome: GenomeVersion
+    reference_genome: Literal[GenomeVersion.HG38] = GenomeVersion.HG38
```

My understanding is that if we get a request specifying it to be HG19 instead, then we would return an error with this change? Is that what we want or do we just want to override such a value with 38?

---

There will be no assignment of that variable anymore in the request. The field is removed from the order portal [here](https://github.com/Clinical-Genomics/clinical-genomics-ui/pull/701) and the column will not be in the new order form. However, based on what you say it would be safer to fix it to this value

---

BUt is there a way in which the assignment doe snot raise an error?

---

When writing this validation flow, quite a lot of effort went into decoupling it from the order portal (not completely but quite a bit). I would prefer if we do not introduce logic which hinges on us knowing how the order portal works and if cURLing towards a swagger gave us the same behaviour. 

It might be that we want to raise an error if the request specifies something other than hg38 though. Just wanted us to think about this.

---

We could in theory remove it as a field or mark it as a private attribute, which we only include in the model_dump

---

We still need to populate the database with this value (that is part of the acceptance criteria), so I think we should keep it somehow. Keeping it hidden is an option too, but I prefer having the fewer hidden arguments as possible

---

I think I agree. These models do dictate what user input is allowed, and the reference genome does not seem to be part of the customer input for Tomte anymore. 

Ideally we would just have a TomteStoringService where we could set the reference genome and remove it completely from here, but now we still have the CaseStoringService. When should the reference genome be set for the Tomte samples now? So far we have sort of made due with private attributes, but I guess there are drawbacks there as well.

---

I will remove it